pretyping/pretyping.ml

```diff
@@ -238,7 +238,7 @@ let apply_inference_hook (hook : inference_hook) env sigma frozen = match frozen
 
 let apply_heuristics env sigma fail_evar =
   (* Resolve eagerly, potentially making wrong choices *)
-  let flags = default_flags_of (Typeclasses.classes_transparent_state ()) in
+  let flags = default_flags_of TransparentState.full in
```

Comment:
Why not use the state from the environment instead? I know we have to keep backwards compatibility as much as possible, but I thought that tactics piercing Opaque directives were considered a bug.

Reply:
OK, trying.